### PR TITLE
Mongo Editors:Use Ext JSON (EJSON) instead of JSON

### LINKED ISSUE #534: Fails to upload a record with _id as non-objectId to mongodb
Repro steps:

1. insert a record with a non-objectId `_id` to a collection in mongo shell. It shows in cosmosdb as:

```json
{
  "_id": "HJxU4ap5f",
  "title": "test393",
}
```

2. in cosmosdb, open the record and modify it with another title:

```json
{
  "_id": "HJxU4ap5f",
  "title": "new",
}
```
3. save it in cosmosdb, causing the following error.

Action: cosmosDB.CosmosEditorManager.onDidSaveTextDocument
Error type: Error
Error Message: Argument passed in must be a single String of 12 bytes or a string of 24 hex characters

Version: 0.6.0
OS: win32

### LINKED ISSUE #564: [Document view] : Need to correctly handle ObjectID-like data when specified in nested fields
Courtesy @Danlock for pointing this out. Pasting comment below: 

Prior discussion [here](https://github.com/Microsoft/vscode-cosmosdb/issues/534#issuecomment-383963490)

Related to this issue, nested objectId are not set as object ids properly, in the case of:

```
{
  "_id": "5aa0150e1430142dda49b551",
  "createdAt": "2018-03-07T16:36:30.436Z",
  "data": {
    "Quasi": "8590406b-67fa-4e08-819d-2860f5fdca79"
  },
  "name": "g6d-car",
  "sensors": [
    {
      "_id": "5adf3b43194dd500010542ea",
      "createdAt": "2018-04-24T14:12:19.693Z",
      "type": "camera"
    }
  ],
  "type": "car",
  "updatedAt": "2018-04-24T14:12:19.693Z"
}
```

If I make any changes the inner sensor ids will be overwritten with "5adf3b43194dd500010542ea" instead of ObjectId("5adf3b43194dd500010542ea").

I can make another issue if you want, but this seems relevant.

### PR DESCRIPTION
**Pull request by PrashanthCorp**
Fixes #564 
Fixes #534 

Tested: 
ObjectID, nested ObjectID array, ISODate, sampleTime, minKey, MaxKey, undefined, numberLong, numberDecimal.
More types [here](https://docs.mongodb.com/manual/reference/mongodb-extended-json/#bson-data-types-and-associated-representations)

### CODE REVIEW

File: src/mongo/editors/MongoCollectionNodeEditor.ts [hunk 1]
```diff
@@ -11,8 +11,10 @@ import { getNodeEditorLabel } from '../../utils/vscodeUtils';
 
 export class MongoCollectionNodeEditor implements ICosmosEditor<IMongoDocument[]> {
     private _collectionNode: IAzureParentNode<MongoCollectionTreeItem>;
+    private EJSON;
     constructor(collectionNode: IAzureParentNode<MongoCollectionTreeItem>) {
         this._collectionNode = collectionNode;
+        this.EJSON = require('mongodb-extended-json');
```
Comment (StephenWeatherford) [inline src/mongo/editors/MongoCollectionNodeEditor.ts]: Why aren't you using import?

Comment (PrashanthCorp) [reply src/mongo/editors/MongoCollectionNodeEditor.ts]: mongodb-extended-JSON doesn't have typings (no index.d.ts, or an entry in the [@types](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types) repo) 

This was the `non-ideal` code I was talking about yesterday

Comment (StephenWeatherford) [reply src/mongo/editors/MongoCollectionNodeEditor.ts]: doesn't mean you can't import at the top:

import * as EJSON = require("...");

Comment (PrashanthCorp) [reply src/mongo/editors/MongoCollectionNodeEditor.ts]: The above seems to throw a ts error. 
`import EJSON = require('mongodb-extended-json')` throws a typings error. 

After discussing with @EricJizbaMSFT , I'm going with `const EJSON = require('mongodb-extended-json')`

**Comment by PrashanthCorp** [comment]
Decisions to use this library outlined in #553

**Pull request merged**